gpt_index/schema.py

```diff
@@ -30,6 +32,9 @@ def __post_init__(self) -> None:
         # assign doc_id if not set
         if self.doc_id is None:
             self.doc_id = get_new_id(set())
+        if self.doc_hash is None:
+            doc_identity = str(self.text) + str(self.extra_info)
```

Comment:
nit: would be worth defining this as its own function for clarity + future re-use (`_generate_doc_hash`)